internal/resources/diffValidation/resource_nsxt_lb_profile.go

```diff
@@ -0,0 +1,168 @@
+//  (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package diffvalidation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	profiles        = "profile_type"
+	http            = "http_profile"
+	tcp             = "tcp_profile"
+	udp             = "udp_profile"
+	cookieProfile   = "cookie_profile"
+	sourceIpProfile = "sourceip_profile"
+	genericProfile  = "generic_profile"
+	clientProfile   = "client_profile"
+	serverProfile   = "server_profile"
+
+	LBHttpProfile                = "LBHttpProfile"
+	LBFastTcpProfile             = "LBFastTcpProfile"
+	LBFastUdpProfile             = "LBFastUdpProfile"
+	LBCookiePersistenceProfile   = "LBCookiePersistenceProfile"
+	LBSourceIpPersistenceProfile = "LBSourceIpPersistenceProfile"
+	LBGenericPersistenceProfile  = "LBGenericPersistenceProfile"
+	LBClientSslProfile           = "LBClientSslProfile"
+	LBServerSslProfile           = "LBServerSslProfile"
+
+	applicationProfile = "application-profile"
+	persistenceProfile = "persistence-profile"
+	sslProfile         = "ssl-profile"
+)
+
+type LoadBalancerProfile struct {
+	diff *schema.ResourceDiff
+}
+
+func NewLoadBalancerProfileValidate(diff *schema.ResourceDiff) *LoadBalancerProfile {
+	return &LoadBalancerProfile{
+		diff: diff,
+	}
+}
+
+func (l *LoadBalancerProfile) DiffValidate() error {
+	err := l.validateProfile()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *LoadBalancerProfile) validateAppProfile(http string, tcp string, udp string) error {
+	h := l.diff.Get(http)
+	t := l.diff.Get(tcp)
```

Comment:
please do not use variable names like this. these would again lead to lint error. give a proper name

Reply:
updated